instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/sessionreplay/ProcessImportanceExt.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sdk.integration.sessionreplay
+
+import android.app.ActivityManager
+import com.datadog.android.rum.DdRumContentProvider
+
+/**
+ * In instrumentation tests, the process is treated as [IMPORTANCE_FOREGROUND_SERVICE],
+ * which differs from the behavior in a typical user environment. To prevent this difference
+ * from affecting test results, this function can be called to override the relevant variable
+ * inside [DdRumContentProvider], allowing the SDK to correctly identify that the application
+ * is in the foreground.
+ */
+fun overrideProcessImportance() {
```

Comment:
This is a nice finding. I wonder how many integration tests are affected by this :(. Good that we fixed this.

Reply:
Actually a lot, it's just in Session Replay test rule we didn't override it, but in `KioskTrackingActivityTestRule` and `GesturesTrackingActivityTestRule` we have already made this override.